packages/block-logs-stream/src/fetchLogs.ts

```diff
@@ -90,7 +90,18 @@ export async function* fetchLogs<TAbiEvents extends readonly AbiEvent[]>({
     try {
       const toBlock = fromBlock + blockRange;
       debug("getting logs", { fromBlock, toBlock });
-      const logs = await publicClient.getLogs({ ...getLogsOpts, fromBlock, toBlock, strict: true });
+
+      const [latestBlockNumber, logs] = await Promise.all([
```

Comment:
This only works if `batch` is enabled in the HTTP transport. This is decided by clients though, so maybe this is put into docs?

Reply:
I wonder if we should have an option to enable the "validate unsynchronized nodes" behavior, where we require `batch` to be enabled or, ideally, do the batch call here directly rather than relying on some client configuration.

Since this approach isn't useful unless the call is atomic (going to the same RPC) and I suspect not all RPC providers support batching, seems like we should put this behind a flag to avoid doubling RPC calls for syncing data from RPC. Could make this the default approach with a way to opt out of this new behavior.